morebits.js

```diff
@@ -3614,10 +3654,36 @@ Morebits.wiki.page = function(pageName, status) {
 		var page = response.pages[0], rev;
 		ctx.pageExists = !page.missing;
 		if (ctx.pageExists) {
-			rev = page.revisions[0];
-			ctx.lastEditTime = rev.timestamp;
-			ctx.pageText = rev.content;
 			ctx.pageID = page.pageid;
+			ctx.lastTouchedTime = page.touched; // Used as basetimestamp when saving
+			// Only actually required for revert editMode, but used for
+			// edit conflict detection and accessible to all via getCurrentID
+			ctx.latestRevID = page.lastrevid;
+
+			rev = page.revisions[0];
+			if (ctx.editMode === 'revert') {
+				// Is this ever even possible?
+				if (rev.revid !== ctx.revertOldID) {
+					ctx.statusElement.error('The retrieved revision does not match the requested revision.');
+					ctx.onLoadFailure(this);
+					return;
+				}
```

Comment:
Is it?  There was a check for this in fluff, but I can't say I've seen or heard of it happening.  At the moment, this is the only reason we [still need to query](https://github.com/wikimedia-gadgets/twinkle/pull/1324/files#diff-20b1ff1a748addb118304a1b428701029480dc64e319f65a82cd213ea3dc8083R2602) for `rvprop=ids` when `revert`ing.